### DISCUSSION TITLE
Save stl data into image files (jpg, png,...)

### DISCUSSION THREAD

**Discussion by eunzihong** [bwoodsend/vtkplotlib#17] [Q&A]
Hi and thank you for your great codes!

Can you tell me how can I save the rendered stl data into image files and set the camera position with some example codes?

Thankyou 😃

**Comment by bwoodsend** [comment]
[save_fig()](https://vtkplotlib.readthedocs.io/en/latest/Figures.html#save-fig) and [view()](https://vtkplotlib.readthedocs.io/en/latest/Figures.html#view) are the functions for saving screenshots and setting the camera angle.
```python
import vtkplotlib as vpl

fig = vpl.figure()
fig.background_opacity = 0  # optional, hide the background in screenshots
vpl.mesh_plot(vpl.data.get_rabbit_stl(), fig=fig)

vpl.view(focal_point=[0, 0, 0], camera_position=[-1, -1, 0], fig=fig)
fig.reset_camera()
vpl.save_fig("screenshot.png", pixels=(1000, 1000), fig=fig)

vpl.close()
```
How you go about choosing the parameters for `view()` depends on your data and what you're doing. If you know the orientation of whatever it is that you're drawing then you can hardcode them like the example above does. If you don't then you'll have to find a way to derive orientation unit vectors from your data.

**Reply by jbernalr** [reply]
I wonder if there is a way to display the actual focal_point and camera_position while on the rendered view seen with `vlp.show()`. That way one could read a setup that is good to generate pngs etc. 
are the conventions regarding the meaning of focal_point and camera_position documented somewhere? 
Thanks!

**Reply by bwoodsend** [reply]
The parameters to `view()` are documented in the docs for [view()](https://vtkplotlib.readthedocs.io/en/latest/Figures.html#view). (I'd be a bit worried if you expected them to be anywhere else.) Calling `view()` outputs its current state so running it after `vpl.show()` will give you the camera positioning/orientation that you left it with.
```python
fig.show()  # Manually set the position to a good angle then close the window
print(vpl.view(fig=fig))  # Use whatever this outputs as your settings to `vpl.view()`
```

**Reply by jbernalr** [reply]
Thanks, that is what I wanted. I did not mean the arguments of view, but the general convention regarding focal point and camera view. But I suppose that is straight forward enough not to be documented. In a cartesian reference system just set the lens of your camera at `camera_position` heading to the global origin and the `focal_point` is then then an extra delta from the 'camera_position'. In any case thanks!
Im using this to fast preview stl files and create little animations
